assets/js/components/data-table.js

```diff
@@ -51,6 +54,7 @@ export const getDataTableFromData = ( data, headers, options ) => {
 		const cells = [];
 		const link = links && links[ j ];
 		const permaLink = link && link[ 0 ] === '/' ? getFullURL( global._googlesitekitLegacyData.admin.siteURL, link ) : link;
+		const adminURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard', { permaLink } ) );
```

Comment:
I'm surprised this doesn't trigger a `rules-of-hooks` ESLint error, but this looks like it would [break the rules of hooks](https://reactjs.org/docs/hooks-rules.html) by calling a hook conditionally and/or a variable number of times. Hooks need to be called in the exact same order every time for every render, and the number of hooks called in each component should not change between renders. Placing hooks inside a loop will cause bugs/errors.

Usually in these cases, we'll want to abstract out a new, small component that selects the `adminURL` based on a `permaLink` prop. In this case, something like replacing `<PrimaryLink>` on line 77 with `<PrimaryAdminLink permaLink={permaLink}>{cell}</PrimaryAdminLink>`. That component would then call the hook, eg:

```js
const PrimaryAdminLink = ({children, permaLink}) => {
    const adminURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard', { permaLink } ) );

    return (
        <PrimaryLink
            className="googlesitekit-table__body-item-link"
            href={ useAdminURLs ? getSiteKitAdminURL( 'googlesitekit-dashboard', { permaLink } ) : link }
            href={ useAdminURLs ? adminURL : link }
            external={ ! useAdminURLs }
            inherit
        >
            { children }
        </PrimaryLink>
    );
};
```

Reply:
Thanks @tofumatt.  No this didn't trigger an ESLint error.  I have refactored this aspect as suggested 👍